tools/richlist.js

```diff
@@ -0,0 +1,559 @@
+#!/usr/bin/env node
+/**
+ * Tool for calculating richlist by hackyminer
+ */
+
+var _ = require('lodash');
+var Web3 = require('web3');
+var async = require('async');
+var BigNumber = require('bignumber.js');
+var mongoose = require('mongoose');
+
+var Account = require('../db.js').Account;
+var Transaction = require('../db.js').Transaction;
+var Block = require('../db.js').Block;
+
+mongoose.connect(process.env.MONGO_URI || 'mongodb://localhost/blockDB');
+
+const ADDRESS_CACHE_MAX = 10000; // address cache threshold
+
+function makeRichList(toBlock, blocks, updateCallback) {
+  var self = makeRichList;
+  if (!self.cached) {
+    self.cached = {};
+    self.index = 0;
+  }
+  if (!self.accounts) {
+    self.accounts = {};
+  }
+  var fromBlock = toBlock - blocks;
+  if (fromBlock < 0) {
+    fromBlock = 0;
+  }
+
+  console.log('Scan accounts from ' + fromBlock + ' to ' + toBlock + ' ...');
+
+  var ended = false;
+  if (fromBlock == toBlock) {
+    ended = true;
+  }
+
+  async.waterfall([
```

Comment:
there are two different account aggregators, one is for normal geth or classic-geth,
the other is for the parity (`--fat-db` enabled case)

aggregate `from addresses` + `to addresses` + `miner addresses` into accounts arrary

Reply:
Gotcha, so this one is for geth/geth-classic aggregator. If you don't mind, can you add a comment about line 20 for `function makeRichList` saying `// RichList for Geth Classic`. The other function `function makeParityRichList` is very clear about what's it for, so that's fine.